src/widgets/immunizations/immunizations-form.component.tsx

```diff
@@ -0,0 +1,348 @@
+                </div>
+              </div>
+              <div className={styles.immunizationsInputContainer}>
+                <label htmlFor="lotNumber">
+                  {t("lot number", "Lot Number")}
+                </label>
+                <div className="omrs-input-group">
+                  <input
+                    className="omrs-input-outlined"
+                    type="number"
+                    data-testid="lotNumberInput"
+                    style={{ height: "2.75rem" }}
+                    defaultValue={lotNumber}
+                    onChange={evt => setLotNumber(evt.target.value)}
+                  />
+                </div>
+              </div>
+              <div className={styles.immunizationsInputContainer}>
+                <label htmlFor="manufacturer">
+                  {t("manufacturer", "Manufacturer")}
+                </label>
+                <div className="omrs-input-group">
+                  <input
+                    className="omrs-input-outlined"
+                    type="text"
+                    data-testid="manufacturerInput"
+                    style={{ height: "2.75rem" }}
+                    defaultValue={manufacturer}
+                    onChange={evt => setManufacturer(evt.target.value)}
+                  />
+                </div>
+              </div>
+            </div>
+          </div>
+        </SummaryCard>
+        <div
+          className={
+            enableCreateButtons || enableEditButtons
+              ? `${styles.buttonStyles} ${styles.buttonStylesBorder}`
+              : styles.buttonStyles
+          }
+        >
+          <button
+            type="button"
+            className="omrs-btn omrs-outlined-neutral omrs-rounded"
+            style={{ width: "50%" }}
+            onClick={closeForm}
+          >
+            {t("cancel", "Cancel")}
+          </button>
+          <button
+            type="submit"
+            style={{ width: "50%" }}
+            className={
+              enableCreateButtons || enableEditButtons
+                ? "omrs-btn omrs-filled-action omrs-rounded"
+                : "omrs-btn omrs-outlined omrs-rounded"
+            }
+            disabled={viewEditForm ? !enableEditButtons : !enableCreateButtons}
+          >
+            {t("save", "Save")}
+          </button>
+        </div>
+      </form>
+    );
+  }
+
+  const closeForm = event => {
+    let userConfirmed: boolean = false;
+    if (formChanged) {
+      userConfirmed = confirm(
+        "There is ongoing work, are you sure you want to close this tab?"
+      );
+    }
+
+    if (userConfirmed && formChanged) {
+      props.entryCancelled();
+      history.push(`/patient/${patientUuid}/chart/immunizations`);
+      props.closeComponent();
+    } else if (!formChanged) {
+      props.entryCancelled();
+      history.push(`/patient/${patientUuid}/chart/immunizations`);
+      props.closeComponent();
+    }
+  };
+
+  const onDoseSelect = event => {
```

Comment:
Can you put that up or make it a `function` to use hoisting here? Thanks! (also either type the `event` or use it as an anonymous lambda above, where `event` would be inferred).